### PR TITLE
:seedling:Simplifying and updating clusterctl upgrade tests

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
Simplifying and updating clusterctl upgrade tests. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by mboukhalfa** [comment]
test-centos-e2e-integration-main

File: test/e2e/upgrade_clusterctl_test.go [hunk 1]
```diff
@@ -29,84 +29,94 @@ var _ = Describe(fmt.Sprintf("When testing cluster upgrade from releases %s > cu
...
+			BootstrapClusterProxy:           bootstrapClusterProxy,
+			ArtifactFolder:                  artifactFolder,
+			SkipCleanup:                     skipCleanup,
+			InitWithCoreProvider:            "cluster-api:v1.6.1",
+			InitWithBootstrapProviders:      []string{"kubeadm:v1.6.1"},
+			InitWithControlPlaneProviders:   []string{"kubeadm:v1.6.1"},
+			InitWithInfrastructureProviders: []string{"metal3:v1.6.0"},
+			InitWithKubernetesVersion:       e2eConfig.GetVariable("INIT_WITH_KUBERNETES_VERSION"),
+			WorkloadKubernetesVersion:       e2eConfig.GetVariable("INIT_WITH_KUBERNETES_VERSION"),
+			InitWithBinary:                  "https://github.com/kubernetes-sigs/cluster-api/releases/download/vv1.6.1/clusterctl-{OS}-{ARCH}",
```
Comment (kashifest) [inline test/e2e/upgrade_clusterctl_test.go:45]:
```suggestion
			InitWithBinary:                  "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.1/clusterctl-{OS}-{ARCH}",
```

**Comment by adilGhaffarDev** [comment]
the test is passed here: https://jenkins.nordix.org/job/metal3_capm3_main_clusterctl_upgrade_test_ubuntu/13/
let's merge the final project infra PR: https://github.com/metal3-io/project-infra/pull/673 , and run the test one final time.

File: scripts/environment.sh [hunk 2]
```diff
@@ -64,7 +64,7 @@ export PATH=$PATH:$HOME/.krew/bin
 
 # Upgrade test environment vars and config
 if [[ ${GINKGO_FOCUS:-} == "clusterctl-upgrade" ]]; then
-  export NUM_NODES=${NUM_NODES:-"5"}
+  export NUM_NODES="5"
```
Comment (kashifest) [inline scripts/environment.sh:67]: why are we hardcoding it?

File: test/e2e/config/e2e_conf.yaml [hunk 3]
```diff
@@ -17,85 +17,96 @@ providers:
 - name: cluster-api
   type: CoreProvider
   versions:
-  - name: ${CAPI_FROM_RELEASE}
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_FROM_RELEASE}/core-components.yaml"
+  - name: v1.6.1
```
Comment (kashifest) [inline test/e2e/config/e2e_conf.yaml:20]: 1.6.3?

File: test/e2e/config/e2e_conf.yaml [hunk 4]
```diff
@@ -17,85 +17,96 @@ providers:
...
+    - old: --metrics-addr=127.0.0.1:8080
+      new: --metrics-addr=:8080
+    files:
+    - sourcePath: "../data/shared/v1.5/metadata.yaml"
 - name: kubeadm
   type: BootstrapProvider
   versions:
-  - name: ${CAPI_FROM_RELEASE} # latest published release in the v1alpha4 series; this is used for v1alpha4 --> v1beta1 clusterctl upgrades test only.
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_FROM_RELEASE}/bootstrap-components.yaml"
+  - name: v1.6.1 # latest published release in the v1alpha4 series; this is used for v1alpha4 --> v1beta1 clusterctl upgrades test only.
```
Comment (kashifest) [inline test/e2e/config/e2e_conf.yaml:41]: Same should it be 1.6.3, also I am removing the comment in my PR

File: test/e2e/config/e2e_conf.yaml [hunk 5]
```diff
@@ -17,85 +17,96 @@ providers:
...
     replacements:
     - old: --metrics-addr=127.0.0.1:8080
       new: --metrics-addr=:8080
     files:
-    - sourcePath: "../data/shared/${CAPI_FROM_RELEASE:0:4}/metadata.yaml"
-  - name: ${CAPI_TO_RELEASE}
-    # Use manifest from source files
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_TO_RELEASE}/control-plane-components.yaml"
+    - sourcePath: "../data/shared/v1.6/metadata.yaml"
+  - name: v1.5.5 # latest published release in the v1alpha4 series; this is used for v1alpha4 --> v1beta1 clusterctl upgrades test only.
```
Comment (kashifest) [inline test/e2e/config/e2e_conf.yaml:71]: Same here why not the latest 1.5.x?

File: scripts/environment.sh (hunk 2 above)
Comment (adilGhaffarDev) [reply scripts/environment.sh:67]: for `clusterctl-upgrade` it is always 5. I think it's ok to hard-code.

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

File: scripts/environment.sh (hunk 2 above)
Comment (kashifest) [reply scripts/environment.sh:67]: true but my point is this script is not used for clusterctl-upgrade only

Comment (adilGhaffarDev) [reply scripts/environment.sh:67]: only when GINKGO_FOCUS=="clusterctl-upgrade" in that scenario we are hardcoding it to 5.

Comment (kashifest) [reply scripts/environment.sh:67]: true,

/approve

**Comment by kashifest** [comment]
@adilGhaffarDev can we take this in?

**Comment by adilGhaffarDev** [comment]
> @adilGhaffarDev can we take this in?

yes. /unhold

File: test/e2e/config/e2e_conf.yaml (hunk 5 above)
Comment (adilGhaffarDev) [reply test/e2e/config/e2e_conf.yaml]: fixed

File: test/e2e/config/e2e_conf.yaml (hunk 4 above)
Comment (adilGhaffarDev) [reply test/e2e/config/e2e_conf.yaml]: fixed

File: test/e2e/config/e2e_conf.yaml (hunk 3 above)
Comment (adilGhaffarDev) [reply test/e2e/config/e2e_conf.yaml]: fixed

File: test/e2e/config/e2e_conf.yaml [hunk 6]
```diff
@@ -17,85 +17,96 @@ providers:
 - name: cluster-api
   type: CoreProvider
   versions:
-  - name: ${CAPI_FROM_RELEASE}
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_FROM_RELEASE}/core-components.yaml"
+  - name: v1.6.3
```
Comment (tuminoid) [inline test/e2e/config/e2e_conf.yaml:20]: We have already bumped main to CAPI 1.7.0-beta.0, should it be added here too?

Comment (adilGhaffarDev) [reply test/e2e/config/e2e_conf.yaml:20]: let's add a separate PR for that, to add an upgrade test for beta it needs more changes.

Comment (tuminoid) [reply test/e2e/config/e2e_conf.yaml:20]: Works for me.

**Pull request merged**